### PR TITLE
LibWeb: Actually run UTF-8 decode without BOM for URLSearchParams

### PR DESCRIPTION
**Pull request by shannonbooth**
Fixes a crash running https://wpt.live/url/urlencoded-parser.any.html

Which, along with the fix cherry-picked here from https://github.com/LadybirdBrowser/ladybird/pull/878 that I just today noticed (important to also prevent conflicts), makes all of the tests from that link pass.

### CODE REVIEW

File: AK/String.h [hunk 1]
```diff
@@ -51,6 +51,10 @@ class String : public Detail::StringBase {
     // Creates a new String from a sequence of UTF-8 encoded code points.
     static ErrorOr<String> from_utf8(StringView);
 
+    // Creates a new String using the replacement character for invalid bytes
+    [[nodiscard]] static String from_bytes_with_replacement(ReadonlyBytes);
+    [[nodiscard]] static String from_bytes_with_replacement(StringView);
```
Comment (awesomekling) [inline AK/String.h:56]: Hmm, how about `from_utf8_with_replacement_character` just to make it ultra explicit?

Comment (shannonbooth) [reply AK/String.h:56]: Sounds good!

Comment (shannonbooth) [reply AK/String.h:56]: Although, the `from_utf` is potentially a bit confusing since it can take something that isn't valid utf8

Comment (awesomekling) [reply AK/String.h:56]: Right, but isn't the idea that it does expect UTF-8, but handles invalid characters by substitution rather than failure?

Comment (shannonbooth) [reply AK/String.h:56]: Right! Yeah I thought about it some more and indeed that seems much more clear :^)

File: Userland/Libraries/LibWeb/DOMURL/URLSearchParams.cpp [hunk 2]
```diff
@@ -115,8 +115,8 @@ ErrorOr<Vector<QueryParam>> url_decode(StringView input)
         auto space_decoded_name = name.replace("+"sv, " "sv, ReplaceMode::All);
 
         // 5. Let nameString and valueString be the result of running UTF-8 decode without BOM on the percent-decoding of name and value, respectively.
-        auto name_string = TRY(String::from_byte_string(URL::percent_decode(space_decoded_name)));
-        auto value_string = TRY(String::from_byte_string(URL::percent_decode(value)));
+        auto name_string = String::from_utf8_with_replacement_character(URL::percent_decode(space_decoded_name));
+        auto value_string = String::from_utf8_with_replacement_character(URL::percent_decode(value));
```
Comment (trflynn89) [inline Userland/Libraries/LibWeb/DOMURL/URLSearchParams.cpp]: I think we're supposed to use LibTextCodec here (namely `UTF8Decoder::to_utf8`). That's our "running UTF-8 decode on bytes" implementation. (And that implementation has a fast path for already-valid UTF-8, which will be most of the web)

**Pull request merged**